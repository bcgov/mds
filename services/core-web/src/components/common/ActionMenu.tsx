import { Button, Dropdown } from "antd";
import { CARAT } from "@/constants/assets";
import React, { FC } from "react";
import { ITableAction } from "@/components/common/CoreTableCommonColumns";

interface ActionMenuProps {
  record: any;
  actionItems: ITableAction[];
  category: string;
}
const ActionMenu: FC<ActionMenuProps> = ({ record, actionItems, category }) => {
  const items = actionItems.map((action) => {
    return {
      key: action.key,
      icon: action.icon,
      label: (
        <button
          type="button"
          className={`full actions-dropdown-button`}
          onClick={(event) => action.clickFunction(event, record)}
        >
          <div>{action.label}</div>
        </button>
      ),
    };
  });
  return (
    <Dropdown menu={{ items }} placement="bottomLeft">
      <Button
        icon={
          <img
            className="padding-sm--right icon-svg-filter"
            src={CARAT}
            alt={`${category} Actions`}
            style={{ paddingLeft: "5px" }}
          />
        }
        type="text"
        className="permit-table-button"
      >
        Actions
      </Button>
    </Dropdown>
  );
};

export default ActionMenu;
