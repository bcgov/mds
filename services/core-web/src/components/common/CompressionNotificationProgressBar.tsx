import React, { FC } from "react";
import { Progress } from "antd";

interface CompressionNotificationProgressBarProps {
  compressionProgress: number;
  notificationTopPosition: number;
}

export const CompressionNotificationProgressBar: FC<CompressionNotificationProgressBarProps> = (
  props
) => (
  <Progress
    percent={props.compressionProgress}
    showInfo={false}
    strokeColor={"#5e46a1"}
    strokeLinecap={"square"}
    trailColor="#d9d9d9"
    className="compressionProgressBar"
    style={{
      top: `${props.notificationTopPosition - 10}px`,
    }}
  />
);

export default CompressionNotificationProgressBar;
