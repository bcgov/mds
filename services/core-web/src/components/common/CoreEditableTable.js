import React from "react";
import { connect } from "react-redux";
import { PropTypes } from "prop-types";
import { getNoticeOfWorkUnitTypeOptionsHash } from "@common/selectors/staticContentSelectors";
import { Field, FieldArray } from "redux-form";
import { Button } from "antd";
import * as Strings from "@common/constants/strings";
import { TRASHCAN } from "@/constants/assets";
import "@ant-design/compatible/assets/index.css";

const propTypes = {
  isViewMode: PropTypes.bool.isRequired,
  fieldName: PropTypes.string.isRequired,
  tableContent: PropTypes.objectOf(PropTypes.any).isRequired,
  fieldID: PropTypes.string.isRequired,
  type: PropTypes.string,
  unitTypeHash: PropTypes.objectOf(PropTypes.string).isRequired,
};

const defaultProps = {
  type: "Activity",
};

const addActivity = (fields) => {
  fields.push({});
};

const removeActivity = (fields, index, fieldID) => {
  if (fields.get(index) && fields.get(index)[fieldID]) {
    // add state_modified and set to "delete" for backend
    fields.get(index).state_modified = "delete";

    // move updated object, this will cause rerendering of the react component, setTimeout is required to bypass react optimization
    setTimeout(() => {
      // eslint-disable-next-line no-constant-condition
      const res = fields.move(index, (index = 0 ? index + 1 : index - 1));
      return res;
    }, 1);
  } else {
    fields.remove(index);
  }
};

const renderActivities = ({ fields, isViewMode, tableContent, type, fieldID, unitTypeHash }) => {
  // resets deleted state if users decided to cancel their changes
  if (isViewMode && fields.length !== 0) {
    fields.getAll().forEach((activity) => {
      // eslint-disable-next-line no-prototype-builtins
      if (activity && activity.hasOwnProperty("state_modified")) {
        delete activity.state_modified;
      }
    });
  }

  const activeRecordsCount =
    fields.length !== 0 ? fields.getAll().filter((activity) => !activity.state_modified).length : 0;

  const renderViewField = (fieldObj, index, content) => {
    const activityObj = fieldObj.get(index);
    if (content.hasUnit) {
      const unit = activityObj[`${content.value}_unit_type_code`]
        ? unitTypeHash[activityObj[`${content.value}_unit_type_code`]]
        : Strings.EMPTY_FIELD;
      return `${activityObj[content.value]} ${unit}`;
    }
    return activityObj[content.value];
  };

  return (
    <div>
      <div className="ant-table-wrapper" style={{ position: "relative", zIndex: 5 }}>
        <div className={`ant-table ${(!fields || fields.length <= 0) && "ant-table-empty"}`}>
          <div className="ant-table-content">
            <table style={{ tableLayout: "auto" }}>
              <thead className="ant-table-thead">
                <tr>
                  {tableContent.map(
                    ({ title, isUnit }) =>
                      (!isUnit || !isViewMode) && <th className="ant-table-cell">{title}</th>
                  )}
                  <th className="ant-table-cell" />
                </tr>
              </thead>
              <tbody className="ant-table-tbody">
                {activeRecordsCount > 0 &&
                  fields.map((activity, index) => {
                    const activityObj = fields.get(index);
                    const key = activityObj && (activityObj[fieldID] || index);
                    return (
                      (isViewMode || (activityObj && !activityObj.state_modified)) && (
                        <tr className="ant-table-row ant-table-row-level-0" key={key}>
                          {tableContent.map((content) => {
                            return isViewMode ? (
                              !content.isUnit && (
                                <td className="ant-table-cell">
                                  <div title={content.title}>
                                    {renderViewField(fields, index, content, tableContent)}
                                  </div>
                                </td>
                              )
                            ) : (
                              <td className="ant-table-cell">
                                <div title={content.title}>
                                  <Field
                                    name={`${activity}.${content.value}`}
                                    value={`${activity}.${content.value}`}
                                    component={content.component}
                                    disabled={isViewMode}
                                    validate={content.validate}
                                    data={content.data || []}
                                    minRows={content.minRows}
                                  />
                                </div>
                              </td>
                            );
                          })}
                          {!isViewMode && (
                            <td className="ant-table-cell">
                              <div name="remove" title="remove">
                                <Button
                                  type="primary"
                                  size="small"
                                  onClick={() => removeActivity(fields, index, fieldID)}
                                  ghost
                                >
                                  <img name="remove" src={TRASHCAN} alt="Remove Activity" />
                                </Button>
                              </div>
                            </td>
                          )}
                          {isViewMode && <td className="ant-table-cell" />}
                        </tr>
                      )
                    );
                  })}
                {activeRecordsCount <= 0 && (
                  <tr className="ant-table-placeholder">
                    <td colSpan="100%" className="ant-table-cell" style={{ color: "#bfbfbf" }}>
                      No Data Yet
                    </td>
                  </tr>
                )}
              </tbody>
            </table>
          </div>
        </div>
      </div>
      {!isViewMode && (
        <div>
          <Button type="primary" onClick={() => addActivity(fields)}>
            {`Add ${type}`}
          </Button>
        </div>
      )}
    </div>
  );
};

export const CoreEditableTable = (props) => {
  return (
    <div>
      <FieldArray
        name={props.fieldName}
        component={renderActivities}
        rerenderOnEveryChange
        {...{
          isViewMode: props.isViewMode,
          tableContent: props.tableContent,
          type: props.type,
          fieldID: props.fieldID,
          unitTypeHash: props.unitTypeHash,
        }}
      />
    </div>
  );
};

CoreEditableTable.propTypes = propTypes;
CoreEditableTable.defaultProps = defaultProps;

export default connect(
  (state) => ({
    unitTypeHash: getNoticeOfWorkUnitTypeOptionsHash(state),
  }),
  null
)(CoreEditableTable);
