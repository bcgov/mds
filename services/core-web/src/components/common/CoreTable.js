import React from "react";
import { PropTypes } from "prop-types";
import { Table } from "antd";
import NullScreen from "@/components/common/NullScreen";

/**
 * @constant CoreTable renders react children or a skeleton loading view using the column headers
 * of the tables and 9 loading rows
 *
 * condition = expecting a truthy value to render children, ie "isLoaded"
 * tableHeaders = array of table headers, the skeleton loading will have the same number of columns and same titles
 * isPaginated = true if the table is paginated - this allows the skeleton to be smooth when changing pages
 *
 *
 */

const propTypes = {
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
  dataSource: PropTypes.objectOf(PropTypes.any).isRequired,
  condition: PropTypes.bool.isRequired,
  tableProps: PropTypes.objectOf(PropTypes.any),
};

const defaultProps = {
  tableProps: {},
};

export const CoreTable = (props) => {
  const baseProps = {
    locale: { emptyText: <NullScreen type="no-results" /> },
    pagination: false,
    rowClassName: "fade-in",
    tableLayout: "auto",
  };
  const combinedProps = { ...baseProps, ...props.tableProps };
  const skeletonColumns = props.columns.map((column) => ({
    title: column.title,
    dataIndex: column.title,
    className: column.className,
    width: column.width,
    render: () => <div className={`skeleton-table__loader ${column.className}`} />,
  }));

  return (
    <div>
      {props.condition ? (
        <div>
          <Table {...combinedProps} columns={props.columns} dataSource={props.dataSource} />
        </div>
      ) : (
        <div className="skeleton-table">
          <Table
            align="left"
            pagination={false}
            columns={skeletonColumns}
            dataSource={new Array(props.tableProps.pagination ? 25 : 9).fill({})}
            rowClassName="skeleton-table__row"
          />
        </div>
      )}
    </div>
  );
};

CoreTable.propTypes = propTypes;
CoreTable.defaultProps = defaultProps;

export default CoreTable;
