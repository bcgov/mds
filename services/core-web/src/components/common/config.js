import RenderCheckbox from "./RenderCheckbox";
import RenderGroupCheckbox from "./RenderGroupCheckbox";
import RenderAutocomplete from "./RenderAutoComplete";
import RenderAutoSizeField from "./RenderAutoSizeField";
import RenderCascader from "./RenderCascader";
import RenderDate from "./RenderDate";
import RenderTime from "./RenderTime";
import RenderYear from "./RenderYear";
import RenderField from "./RenderField";
import RenderScrollField from "./RenderScrollField";
import RenderLargeSelect from "./RenderLargeSelect";
import RenderSelect from "./RenderSelect";
import RenderMultiSelect from "./RenderMultiSelect";
import RenderRadioButtons from "./RenderRadioButtons";
import RenderGroupedSelect from "./RenderGroupedSelect";
import RenderMineSelect from "./RenderMineSelect";
import RenderLabel from "./RenderLabel";

// This file is anticipated to have multiple exports
// eslint-disable-next-line import/prefer-default-export
export const renderConfig = {
  CHECKBOX: RenderCheckbox,
  GROUP_CHECK_BOX: RenderGroupCheckbox,
  AUTOCOMPLETE: RenderAutocomplete,
  AUTO_SIZE_FIELD: RenderAutoSizeField,
  CASCADER: RenderCascader,
  DATE: RenderDate,
  TIME: RenderTime,
  YEAR: RenderYear,
  FIELD: RenderField,
  SCROLL_FIELD: RenderScrollField,
  SELECT: RenderSelect,
  LARGE_SELECT: RenderLargeSelect,
  MULTI_SELECT: RenderMultiSelect,
  RADIO: RenderRadioButtons,
  GROUPED_SELECT: RenderGroupedSelect,
  MINE_SELECT: RenderMineSelect,
  LABEL: RenderLabel,
};
