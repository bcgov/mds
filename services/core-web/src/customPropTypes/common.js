import { PropTypes, shape, arrayOf, objectOf, oneOfType } from "prop-types";

export const dropdownListItem = shape({
  value: oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string.isRequired,
});

export const filterListItem = shape({
  value: oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  text: PropTypes.string.isRequired,
});

export const groupedDropdownList = shape({
  groupName: oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  opt: arrayOf(dropdownListItem),
});

export const options = arrayOf(dropdownListItem);
export const groupOptions = arrayOf(groupedDropdownList);
export const filterOptions = arrayOf(filterListItem);

export const formMeta = shape({
  touched: PropTypes.bool,
  error: PropTypes.string,
  warning: PropTypes.string,
});

export const match = shape({
  isExact: PropTypes.bool,
  params: objectOf(PropTypes.string),
  path: PropTypes.string,
  url: PropTypes.string,
});

export const genericFormState = shape({
  anyTouched: PropTypes.bool,
  registeredFields: objectOf(objectOf(oneOfType([PropTypes.string, PropTypes.number]))),
  fields: objectOf(objectOf(PropTypes.bool)),
  syncErrors: PropTypes.string,
  values: objectOf(PropTypes.string),
});

export const pageData = shape({
  current_page: PropTypes.number,
  items_per_page: PropTypes.number,
  records: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  total: PropTypes.number,
  total_pages: PropTypes.number,
});
