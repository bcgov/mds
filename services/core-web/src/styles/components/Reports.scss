/* Minespace is wrapped with antD Row/Col components to control max width, since Core
   isn't, this needed to be added to keep display in line with the design on Core.
 */
.report-steps-page {
  width: 100%;
  margin: 0 auto;
}

@media (min-width: 992px) {
  .report-steps-page {
    max-width: calc(992px * 22 / 24);
  }
}

@media (min-width: 1200px) {
  .report-steps-page {
    max-width: calc(1200px * 20 / 24);
  }
}

@media (min-width: 1600px) {
  .report-steps-page {
    max-width: calc(1600px * 18 / 24);
  }
}

.report-steps-page h5 {
  text-transform: none !important;
  font-weight: bold;
}

.report-title {
  margin-bottom: 36px !important;
  margin-top: 16px;
}

.report-steps {
  margin-bottom: 36px;
  margin-top: 36px;
}

.report-link {
  text-decoration: none;
  line-height: 32px;
  border: none !important;
  margin: 0 !important;
}

.report-instructions {
  color: rgba(49, 49, 50, 0.55);
}

.light-grey-border {
  border: 1px solid $background;
  padding: 16px;
}

.report-info-box {
  height: 100%;
  padding: 16px;
  border: 1px solid $light-violet;

  background: #fff;

  h5 {
    text-transform: none;
    font-weight: bold;
  }
}

.light-grey-background {
  background-color: $light-grey;
}