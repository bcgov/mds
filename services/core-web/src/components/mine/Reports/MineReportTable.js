import React from "react";
import PropTypes from "prop-types";
import * as Strings from "@common/constants/strings";
import { formatDate, truncateFilename, dateSorter } from "@common/utils/helpers";
import { downloadFileFromDocumentManager } from "@common/utils/actionlessNetworkCalls";
import NullScreen from "@/components/common/NullScreen";
import { COLOR } from "@/constants/styles";
import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
import * as Permission from "@/constants/permissions";
import CustomPropTypes from "@/customPropTypes";
import { MineReportActions } from "@/components/mine/Reports/MineReportActions";
import LinkButton from "@/components/common/LinkButton";
import CoreTable from "@/components/common/CoreTable";

const { errorRed } = COLOR;

/**
 * @class  MinePermitInfo - contains all permit information
 */

const propTypes = {
  mineReports: PropTypes.arrayOf(CustomPropTypes.mineReport).isRequired,
  openEditReportModal: PropTypes.func.isRequired,
  handleEditReport: PropTypes.func.isRequired,
  handleRemoveReport: PropTypes.func.isRequired,
  isLoaded: PropTypes.bool.isRequired,
};

const columns = [
  {
    title: "Report Name",
    dataIndex: "report_name",
    key: "report_name",
    sorter: (a, b) => a.report_name.localeCompare(b.report_name),
    render: (text, record) => (
      <div title="Report Name" style={record.isOverdue ? { color: errorRed } : {}}>
        {record.report_name}
      </div>
    ),
  },
  {
    title: "Compliance Year/Period",
    dataIndex: "submission_year",
    key: "submission_year",
    sorter: (a, b) => (a.submission_year > b.submission_year ? -1 : 1),
    render: (text, record) => (
      <div title="Year" style={record.isOverdue ? { color: errorRed } : {}}>
        {record.submission_year}
      </div>
    ),
  },
  {
    title: "Due",
    dataIndex: "due_date",
    key: "due_date",
    sorter: dateSorter("due_date"),
    render: (text, record) => (
      <div title="Due" style={record.isOverdue ? { color: errorRed } : {}}>
        {formatDate(record.due_date) || Strings.EMPTY_FIELD}
      </div>
    ),
  },
  {
    title: "Requested By",
    dataIndex: "created_by_idir",
    key: "created_by_idir",
    render: (text) => <div title="Requested By">{text}</div>,
  },
  {
    title: "Received",
    dataIndex: "received_date",
    key: "received_date",
    sorter: dateSorter("received_date"),
    render: (text, record) => (
      <div title="Received" style={record.isOverdue ? { color: errorRed } : {}}>
        {formatDate(record.received_date) || Strings.EMPTY_FIELD}
      </div>
    ),
  },
  {
    title: "Documents",
    dataIndex: "documents",
    key: "documents_key",
    render: (text, record) => (
      <div title="Documents" className="cap-col-height">
        {record.report.mine_report_submissions.length > 0 &&
        record.report.mine_report_submissions[record.report.mine_report_submissions.length - 1]
          .documents.length > 0
          ? record.report.mine_report_submissions[
              record.report.mine_report_submissions.length - 1
            ].documents.map((file) => (
              <div key={file.mine_document_guid} title={file.document_name}>
                <LinkButton onClick={() => downloadFileFromDocumentManager(file)}>
                  {truncateFilename(file.document_name)}
                </LinkButton>
              </div>
            ))
          : Strings.EMPTY_FIELD}
      </div>
    ),
  },
  {
    title: "",
    dataIndex: "record",
    render: (text, record) => {
      return (
        <div title="" align="right">
          <AuthorizationWrapper permission={Permission.EDIT_REPORTS}>
            <MineReportActions
              mineReport={record.report}
              openEditReportModal={record.openEditReportModal}
              handleEditReport={record.handleEditReport}
              handleRemoveReport={record.handleRemoveReport}
            />
          </AuthorizationWrapper>
        </div>
      );
    },
  },
];

const transformRowData = (report, openEditReportModal, handleEditReport, handleRemoveReport) => ({
  key: report.report_guid,
  documents_key: `${report.report_guid}_documents`,
  report,
  report_name: report.report_name,
  due_date: report.due_date,
  created_by_idir: report.created_by_idir,
  received_date: report.received_date,
  submission_year: report.submission_year,
  openEditReportModal,
  handleEditReport,
  handleRemoveReport,
});

export const MineReportTable = (props) => (
  <CoreTable
    condition={props.isLoaded}
    columns={columns}
    dataSource={props.mineReports.map((r) =>
      transformRowData(
        r,
        props.openEditReportModal,
        props.handleEditReport,
        props.handleRemoveReport
      )
    )}
    tableProps={{
      align: "left",
      pagination: false,
      locale: { emptyText: <NullScreen type="reports" /> },
    }}
  />
);

MineReportTable.propTypes = propTypes;

export default MineReportTable;
