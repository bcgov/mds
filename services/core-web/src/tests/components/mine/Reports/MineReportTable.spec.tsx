import React from "react";
import { MineReportTable } from "@/components/mine/Reports/MineReportTable";
import * as MOCK from "@/tests/mocks/dataMocks";
import { AUTHENTICATION } from "@mds/common";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

// const dispatchProps = {};
const props = {
  mine: MOCK.MINES.mines[MOCK.MINES.mineIds[0]],
  mineReports: MOCK.MINE_REPORTS,
  mineReportCategoryOptionsHash: MOCK.MINE_REPORT_CATEGORY_OPTIONS_HASH,
  mineReportStatusOptionsHash: MOCK.MINE_REPORT_STATUS_OPTIONS_HASH,
};

// const setupProps = () => {
//   props.mine = MOCK.MINES.mines[MOCK.MINES.mineIds[0]];
//   props.mineReports = MOCK.MINE_REPORTS;
//   props.mineReportCategoryOptionsHash = MOCK.MINE_REPORT_CATEGORY_OPTIONS_HASH;
//   props.mineReportStatusOptionsHash = MOCK.MINE_REPORT_STATUS_OPTIONS_HASH;
// };

// beforeEach(() => {
//   setupProps();
// });

// describe("MineReportTable", () => {
//   it.skip("renders properly", () => {
//     const component = shallow(<MineReportTable {...dispatchProps} {...props} />);
//     expect(component).toMatchSnapshot();
//   });
// });
const initialState = {
  [AUTHENTICATION]: {
    userAccessData: MOCK.USER_ACCESS_DATA,
  },
};
describe("MineReportTable", () => {
  it("renders properly", () => {
    const { container } = render(
      <ReduxWrapper initialState={initialState}>
        <BrowserRouter>
          <MineReportTable {...props} />
        </BrowserRouter>
      </ReduxWrapper>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
