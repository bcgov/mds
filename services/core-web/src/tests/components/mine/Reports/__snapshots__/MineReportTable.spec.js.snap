// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MineReportTable renders properly 1`] = `
<CoreTable
  columns={
    Array [
      Object {
        "className": "column-hide",
        "dataIndex": "mine_name",
        "key": "mine_name",
        "render": [Function],
        "sortField": "mine_name",
        "sortOrder": false,
        "sorter": false,
        "title": "Mine",
      },
      Object {
        "dataIndex": "report_name",
        "key": "report_name",
        "render": [Function],
        "sortField": "report_name",
        "sortOrder": false,
        "sorter": [Function],
        "title": "Report Name",
      },
      Object {
        "dataIndex": "submission_year",
        "key": "submission_year",
        "render": [Function],
        "sortField": "submission_year",
        "sortOrder": false,
        "sorter": [Function],
        "title": "Compliance Year",
      },
      Object {
        "dataIndex": "mine_report_status_code",
        "key": "mine_report_status_code",
        "render": [Function],
        "sortField": "mine_report_status_code",
        "sortOrder": false,
        "sorter": [Function],
        "title": "Status",
      },
      Object {
        "dataIndex": "due_date",
        "key": "due_date",
        "render": [Function],
        "sortField": "due_date",
        "sortOrder": false,
        "sorter": [Function],
        "title": "Due",
      },
      Object {
        "dataIndex": "received_date",
        "key": "received_date",
        "render": [Function],
        "sortField": "received_date",
        "sortOrder": false,
        "sorter": [Function],
        "title": "Received",
      },
      Object {
        "className": "",
        "dataIndex": "created_by_idir",
        "key": "created_by_idir",
        "render": [Function],
        "sortField": "created_by_idir",
        "sortOrder": false,
        "sorter": [Function],
        "title": "Requested By",
      },
      Object {
        "dataIndex": "documents",
        "key": "documents",
        "render": [Function],
        "sortOrder": false,
        "title": "Documents",
      },
      Object {
        "key": "operations",
        "render": [Function],
        "sortOrder": false,
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "created_by_idir": "idir\\\\TEST",
        "documents": Array [
          Object {
            "document_manager_guid": "b660371a-d3fa-41ff-99ed-f9b86dff72f7",
            "document_name": "0101581201901_APPLICATION_FORM (3) (2) (1).pdf",
            "mine_document_guid": "98a9d9de-4feb-41a7-adda-842421462657",
            "mine_guid": "abed555d-7391-49e7-bc04-e2c6062432c6",
            "upload_date": "2020-01-29T20:50:51.439631+00:00",
          },
        ],
        "due_date": "Jan 02 2020",
        "handleEditReport": undefined,
        "handleRemoveReport": undefined,
        "key": "9f98a719-720a-40a5-ac5b-e91e8a526fad",
        "mine_guid": "aa3cb08a-ee1b-4dc9-8bf6-f54eb7484d4d",
        "mine_name": "Abbott Inc",
        "mine_report_category": ", Geotechnical",
        "mine_report_definition_guid": "baa01f9f-c9b2-485d-96f3-12a9c8fe637b",
        "mine_report_guid": "9f98a719-720a-40a5-ac5b-e91e8a526fad",
        "mine_report_id": 123,
        "mine_report_status_code": "Not Requested",
        "openEditReportModal": undefined,
        "permit_guid": "N/A",
        "received_date": "Jan 10 2020",
        "report": Object {
          "created_by_idir": "idir\\\\TEST",
          "due_date": "2020-01-02",
          "mine_guid": "aa3cb08a-ee1b-4dc9-8bf6-f54eb7484d4d",
          "mine_name": "Abbott Inc",
          "mine_report_category": Array [
            "GSC",
            "GTC",
          ],
          "mine_report_definition_guid": "baa01f9f-c9b2-485d-96f3-12a9c8fe637b",
          "mine_report_guid": "9f98a719-720a-40a5-ac5b-e91e8a526fad",
          "mine_report_id": 123,
          "mine_report_submissions": Array [
            Object {
              "comments": Array [],
              "documents": Array [
                Object {
                  "document_manager_guid": "b660371a-d3fa-41ff-99ed-f9b86dff72f7",
                  "document_name": "0101581201901_APPLICATION_FORM (3) (2) (1).pdf",
                  "mine_document_guid": "98a9d9de-4feb-41a7-adda-842421462657",
                  "mine_guid": "abed555d-7391-49e7-bc04-e2c6062432c6",
                  "upload_date": "2020-01-29T20:50:51.439631+00:00",
                },
              ],
              "mine_report_submission_guid": "fed32646-5db7-495a-acbb-b6b8ad333ee1",
              "mine_report_submission_status_code": "NRQ",
              "submission_date": "2020-01-29",
            },
          ],
          "permit_guid": null,
          "received_date": "2020-01-10",
          "report_name": "Underground Oil and Grease Storage Area Report",
          "submission_year": 2020,
        },
        "report_name": "Underground Oil and Grease Storage Area Report",
        "submission_year": 2020,
      },
      Object {
        "created_by_idir": "idir\\\\TEST",
        "documents": Array [
          Object {
            "document_manager_guid": "dde0b5bf-a14d-4272-8a65-467440b01294",
            "document_name": "0101581201901_APPLICATION_FORM (1) (1).pdf",
            "mine_document_guid": "3d4e420f-3f0a-4c54-ad07-e3aeb74cbe89",
            "mine_guid": "abed555d-7391-49e7-bc04-e2c6062432c6",
            "upload_date": "2020-01-29T20:51:53.466101+00:00",
          },
        ],
        "due_date": "Mar 31 2020",
        "handleEditReport": undefined,
        "handleRemoveReport": undefined,
        "key": "b59a166e-749e-4e6c-a232-d4c55f1f227c",
        "mine_guid": "aa3cb08a-ee1b-4dc9-8bf6-f54eb7484d4d",
        "mine_name": "Abbott Inc",
        "mine_report_category": "Geotechnical, Tailings Storage Facility",
        "mine_report_definition_guid": "5f4f4727-4ecd-4a04-8929-2e8a5e03996d",
        "mine_report_guid": "b59a166e-749e-4e6c-a232-d4c55f1f227c",
        "mine_report_id": 124,
        "mine_report_status_code": "Not Requested",
        "openEditReportModal": undefined,
        "permit_guid": "N/A",
        "received_date": "Jan 02 2020",
        "report": Object {
          "created_by_idir": "idir\\\\TEST",
          "due_date": "2020-03-31",
          "mine_guid": "aa3cb08a-ee1b-4dc9-8bf6-f54eb7484d4d",
          "mine_name": "Abbott Inc",
          "mine_report_category": Array [
            "GTC",
            "TSF",
          ],
          "mine_report_definition_guid": "5f4f4727-4ecd-4a04-8929-2e8a5e03996d",
          "mine_report_guid": "b59a166e-749e-4e6c-a232-d4c55f1f227c",
          "mine_report_id": 124,
          "mine_report_submissions": Array [
            Object {
              "comments": Array [],
              "documents": Array [
                Object {
                  "document_manager_guid": "dde0b5bf-a14d-4272-8a65-467440b01294",
                  "document_name": "0101581201901_APPLICATION_FORM (1) (1).pdf",
                  "mine_document_guid": "3d4e420f-3f0a-4c54-ad07-e3aeb74cbe89",
                  "mine_guid": "abed555d-7391-49e7-bc04-e2c6062432c6",
                  "upload_date": "2020-01-29T20:51:53.466101+00:00",
                },
              ],
              "mine_report_submission_guid": "d0149d1b-845d-4011-a731-3f951c7d8219",
              "mine_report_submission_status_code": "NRQ",
              "submission_date": "2020-01-29",
            },
          ],
          "permit_guid": null,
          "received_date": "2020-01-02",
          "report_name": "TSF, WSF or Dam As-built Report",
          "submission_year": 2020,
        },
        "report_name": "TSF, WSF or Dam As-built Report",
        "submission_year": 2020,
      },
      Object {
        "created_by_idir": "idir\\\\TEST",
        "documents": Array [],
        "due_date": "Mar 31 2020",
        "handleEditReport": undefined,
        "handleRemoveReport": undefined,
        "key": "92327cd3-eec0-4e18-b898-25539ac408e9",
        "mine_guid": "aa3cb08a-ee1b-4dc9-8bf6-f54eb7484d4d",
        "mine_name": "Abbott Inc",
        "mine_report_category": "Geotechnical, Tailings Storage Facility",
        "mine_report_definition_guid": "6eda0c36-8748-4072-83c9-0fcdf270d36f",
        "mine_report_guid": "92327cd3-eec0-4e18-b898-25539ac408e9",
        "mine_report_id": 125,
        "mine_report_status_code": null,
        "openEditReportModal": undefined,
        "permit_guid": "N/A",
        "received_date": null,
        "report": Object {
          "created_by_idir": "idir\\\\TEST",
          "due_date": "2020-03-31",
          "mine_guid": "aa3cb08a-ee1b-4dc9-8bf6-f54eb7484d4d",
          "mine_name": "Abbott Inc",
          "mine_report_category": Array [
            "GTC",
            "TSF",
          ],
          "mine_report_definition_guid": "6eda0c36-8748-4072-83c9-0fcdf270d36f",
          "mine_report_guid": "92327cd3-eec0-4e18-b898-25539ac408e9",
          "mine_report_id": 125,
          "mine_report_submissions": Array [],
          "permit_guid": null,
          "received_date": null,
          "report_name": "Annual DSI",
          "submission_year": 2020,
        },
        "report_name": "Annual DSI",
        "submission_year": 2020,
      },
    ]
  }
  tableProps={
    Object {
      "align": "left",
      "locale": Object {
        "emptyText": <NullScreen
          message=""
          type="reports"
        />,
      },
      "onChange": [Function],
      "pagination": false,
    }
  }
/>
`;
